latextools_utils/tex_directives.py

```diff
@@ -0,0 +1,116 @@
+from __future__ import print_function
+
+import codecs
+import re
+import sublime
+import sys
+import traceback
+
+if sys.version_info < (3, 0):
+    strbase = basestring
+else:
+    strbase = str
+
+TEX_DIRECTIVE = re.compile(
+    r'%+\s*!(?:T|t)(?:E|e)(?:X|x)\s+([\w-]+)\s*=\s*' +
```

Comment:
1. whitespaces between `%!` and `TEX` should be allowed, hence `%+\s*!\s*(?:T|t)` instead of `%+\s*!(?:T|t)`
2. You could use the `re.IGNORECASE` flag instead of writing every letter und upper and lower case